### DISCUSSION TITLE
HSL color support

### DISCUSSION THREAD

**Discussion by sandstrom** [typst/typst#2512] [Ideas]
Would be neat if typst supported a HSL color function (in addition to rgb).

HSL is easier to work with, since oftentimes a developer wants to make a color "lighter" or "darker", and doing that with rgb is more work. With HSL it's just a change to one parameter.

Converting from HSL to RGB is easy:
https://docs.rs/colors-transform/latest/colors_transform/

There might also be arguments for introducing support for additional color spaces, but since "sRGB" is already supported (it's used for RGB), adding support for hsl (and potentially also hwb) isn't much work.

**Comment by Enivex** [comment]
I'm fairly sure HSL is already supported on main

**Reply by laurmaedje** [reply]
Yep.

**Reply by sandstrom** [reply]
Nice! I'll look for it in the next release
